relay.go

```diff
@@ -364,10 +369,18 @@ func (r *Relay) handleHopStream(s inet.Stream, msg *pb.CircuitRelay) {
 
 	r.addLiveHop(src.ID, dst.ID)
 
+	goroutines := new(int32)
```

Comment:
note that this is allocated on the heap to avoid lingering the original goroutine stack.